src/dlgTriggerEditor.cpp

```diff
@@ -4523,27 +4523,27 @@ void dlgTriggerEditor::saveVar()
         return;
     }
     mChangingVar = true;
-    int nameType = mpVarsMainArea->comboBox_variable_key_type->itemData(mpVarsMainArea->comboBox_variable_key_type->currentIndex(), Qt::UserRole).toInt();
-    int valueType = mpVarsMainArea->comboBox_variable_value_type->itemData(mpVarsMainArea->comboBox_variable_value_type->currentIndex(), Qt::UserRole).toInt();
-    if ((nameType == 3 || nameType == 4) && newVar) {
-        nameType = -1;
+    int uiNameType = mpVarsMainArea->comboBox_variable_key_type->itemData(mpVarsMainArea->comboBox_variable_key_type->currentIndex(), Qt::UserRole).toInt();
+    int uiValueType = mpVarsMainArea->comboBox_variable_value_type->itemData(mpVarsMainArea->comboBox_variable_value_type->currentIndex(), Qt::UserRole).toInt();
+    if ((uiNameType == 3 || uiNameType == 4) && newVar) {
+        uiNameType = -1;
     }
     //check variable recasting
-    int varRecast = canRecast(pItem, nameType, valueType);
-    if ((nameType == -1) || (var && nameType != var->getKeyType())) {
+    int varRecast = canRecast(pItem, uiNameType, uiValueType);
+    if ((uiNameType == -1) || (var && uiNameType != var->getKeyType())) {
         if (QString(newName).toInt()) {
```

Comment:
I suspect this test may have bogus results - a string of `"0"` will happily convert to the integer value `0` but that will fail this `if` - I think you need to have:

    bool isOk;
    if (QString(newName).toInt(&isOk) || isOk) {

to catch that corner case! Also - it is permissible for the boolean `true` and `false` values to also be valid *key* values? Hey - could something like that explain why zero values always come at the end of a list of 0, 1, 2 , ..., n keys in the output of `display()` when fed a lua arrays? The zero is being converted to a string rather than a number {yeah I know lua arrays begin with 1 not 0...!}

Reply:
I didn't change this code at all as the diff shows, so this isn't related to this PR. As for the display - I doubt it - the function has no interaction with this code, it's all Lua side!